src/physics/exact_solutions/exact_solution.cpp

```diff
@@ -0,0 +1,89 @@
+#include <deal.II/base/function.h>
+#include "exact_solution.h"
+
+namespace PHiLiP {
+
+// ========================================================
+// ZERO -- Returns zero everywhere; used a placeholder when no exact solution is defined.
+// ========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction_Zero<dim,nstate,real>
+::ExactSolutionFunction_Zero(double time_compare)
+        : ExactSolutionFunction<dim,nstate,real>()
+{
+    t = time_compare;
+}
+
+template <int dim, int nstate, typename real>
+inline real ExactSolutionFunction_Zero<dim,nstate,real>
+::value(const dealii::Point<dim,real> &/*point*/, const unsigned int /*istate*/) const
+{
+    real value = 0;
+    return value;
+}
+
+// ========================================================
+// 1D SINE -- Exact solution for advection_explicit_time_study
+// ========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction_1DSine<dim,nstate,real>
+::ExactSolutionFunction_1DSine (double time_compare)
+        : ExactSolutionFunction<dim,nstate,real>()
+{
+    t = time_compare;
+}
+
+template <int dim, int nstate, typename real>
+inline real ExactSolutionFunction_1DSine<dim,nstate,real>
+::value(const dealii::Point<dim,real> &point, const unsigned int /*istate*/) const
+{
+    double x_adv_speed = 1.0;
+
+    real value = 0;
+    real pi = dealii::numbers::PI;
+    if(point[0] >= 0.0 && point[0] <= 2.0){
+        value = sin(2*pi*(point[0] - x_adv_speed * t)/2.0);
+    }
+    return value;
+}
+
+
+//=========================================================
+// FLOW SOLVER -- Exact Solution Base Class + Factory
+//=========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction<dim,nstate,real>
+::ExactSolutionFunction ()
+    : dealii::Function<dim,real>(nstate)
+{
+    //do nothing
```

Comment:
What is the purpose of the base class if it does nothing? If it is just to define a `ExactSolutionFunction` type, you might want to use an abstract class.

Reply:
I used the same structure as `initial_condition.cpp`, which also uses an [empty base class](https://github.com/dougshidong/PHiLiP/blob/b13f5ff03c9ee050d21d1716516908e0b6513072/src/physics/initial_conditions/initial_condition.cpp#L164). I would rather leave this topic to be addressed with Issue #143.

Reply:
Yup, that's what I had previously noticed, it is basically a middle interface to dealii::Function that adds no functionality, just like InitialConditions. Issue #143 will remedy that.

Reply:
Sounds good.